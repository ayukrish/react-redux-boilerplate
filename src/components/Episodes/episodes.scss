.episodeWrapper {
  justify-content: space-around;
  padding: 20px;
}