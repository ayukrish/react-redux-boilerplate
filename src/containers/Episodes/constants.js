export const GET_EPISODES = 'GET_EPISODES';
export const SET_EPISODES = 'SET_EPISODES';
