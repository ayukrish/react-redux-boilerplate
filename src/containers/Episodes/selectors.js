export const episodeData = (state) => state.episodes && state.episodes.data;
