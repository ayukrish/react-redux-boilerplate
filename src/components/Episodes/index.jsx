import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Card from '../Card';
import Pagination from '../Pagination';
import * as styles from './episodes.scss';


const propTypes = {
  getEpisodes: PropTypes.func,
  episodeData: PropTypes.object
};

const defaultProps = {
  episodeData: {},
  getEpisodes: () => { }
};

export default class Episodes extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentPage: 1
    };
  }

  componentDidMount() {
    this.props.getEpisodes(this.state.currentPage);
  }

  render() {
    const { episodeData } = this.props;
    const { results = {} } = episodeData;
    return (
      <section className={`${styles.episodeWrapper} flex wrap`}>
        {results.length > 0 && results.map(item => (
          <Card
            key={item.id}
            dataObj={{
              EPISODE: item && item.episode,
              'AIR DATE': item && item.air_date
            }}
            heading={item && item.name}
          />
        ))}
        <Pagination
          contentLength={(results && results.length) || 0}
          currentPage={this.state.currentPage}
          limit={20}
          onChange={currentPage => {
            this.props.getEpisodes(currentPage);
            this.setState({
              currentPage
            });
          }}
        />
      </section>
    );
  }
}

Episodes.propTypes = propTypes;
Episodes.defaultProps = defaultProps;
