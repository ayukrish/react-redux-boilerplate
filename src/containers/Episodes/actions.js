import { GET_EPISODES, SET_EPISODES } from './constants';


export const getEpisodes = pageNo => ({ type: GET_EPISODES, data: pageNo });

export const setEpisodes = data => ({ type: SET_EPISODES, data });
