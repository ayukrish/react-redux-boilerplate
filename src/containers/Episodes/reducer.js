import { SET_EPISODES } from './constants';

const INITIAL_STATE = {
  data: {}
};

/**
 * Reducer - to update services state
 * @param {Object} state initial/default services state.
 */
export default function (state = INITIAL_STATE, { type, data }) {
  switch (type) {
    case SET_EPISODES:
      return { ...state, data };
    default:
      return state;
  }
}
