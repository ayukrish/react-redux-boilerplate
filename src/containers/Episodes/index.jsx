import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { getEpisodes } from './actions';
import Episodes from '../../components/Episodes';
import { episodeData } from './selectors';

const mapStateToProps = createStructuredSelector({
  episodeData
});

const mapDispatchToProps = (dispatch) => ({
  getEpisodes: (pageNo) => dispatch(getEpisodes(pageNo))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Episodes);
