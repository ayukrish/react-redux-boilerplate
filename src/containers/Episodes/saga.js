import { takeLatest, put, call } from 'redux-saga/effects';
import { service } from '../../service';
import { enpoints } from '../../endpoint';
import { setEpisodes } from './actions';
import { GET_EPISODES } from './constants';

export function* getEpisodesSaga({ data }) {
  try {
    // data is obtained after axios call is resolved
    const response = yield call(service, {
      url: enpoints.episodes,
      method: 'get',
      page: data
    });
    yield put(setEpisodes(response));
  } catch (e) {
    // catch error on a bad axios call
    // alert using an alert library
  }
}

export function* getEpisodeSaga() {
  yield takeLatest(GET_EPISODES, getEpisodesSaga);
}
