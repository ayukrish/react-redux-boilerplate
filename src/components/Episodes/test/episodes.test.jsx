import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

// Components
import Episodes from '../index';


describe('Episodes Test Suite', () => {
  it('renders correctly without any data', () => {
    const output = shallow(
      <Episodes />,
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('renders correctly without any data', () => {
    const output = shallow(
      <Episodes
        getEpisodes={() => { }}
        episodeData={{
          results: [{
            id: 1,
            name: 'Pilot',
            air_date: 'December 2, 2013',
            episode: 'S01E01'
          }]
        }}
      />,
    );
    output.find('Pagination').props().onChange(1);
    expect(output.state().currentPage).toEqual(1);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
