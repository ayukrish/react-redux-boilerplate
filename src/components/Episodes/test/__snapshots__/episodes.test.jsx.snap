// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episodes Test Suite renders correctly without any data 1`] = `
<section
  className="undefined flex wrap"
>
  <Pagination
    contentLength={0}
    currentPage={1}
    limit={20}
    onChange={[Function]}
  />
</section>
`;

exports[`Episodes Test Suite renders correctly without any data 2`] = `
<section
  className="undefined flex wrap"
>
  <Card
    dataObj={
      Object {
        "AIR DATE": "December 2, 2013",
        "EPISODE": "S01E01",
      }
    }
    heading="Pilot"
    imgSrc=""
    key="1"
  />
  <Pagination
    contentLength={1}
    currentPage={1}
    limit={20}
    onChange={[Function]}
  />
</section>
`;
